spacy/tokens/span.pyx

```diff
@@ -653,7 +653,7 @@ cdef class Span:
             return self.doc.vocab.strings[self.label]
 
         def __set__(self, unicode label_):
-            self.label = self.doc.vocab.strings.add(label_)
+            raise NotImplementedError(TempErrors.T007.format(attr="label_"))
```

Comment:
Maybe we want a different error message here? `T007` says "Vote for this feature on the issue tracker", but... do we want this? The implications of writing to a span are pretty subtle and I think we've now come up with a few good reasons why we can't make `Span.label` writable like this. So the error message might be misleading.

Instead, maybe we can add a custom error that explains why writing to the span's label isn't possible? Like, a `Span` is only a view and its label corresponds to the tokens entity labels. If you want to set entities, you should create a new `Span` like this: etc etc.

cc: @honnibal

Reply:
All good points - made the changes accordingly!